core/src/test/java/org/elasticsearch/search/aggregations/metrics/percentiles/hdr/InternalHDRPercentilesRanksTests.java

```diff
@@ -61,4 +66,26 @@ protected void assertReduced(InternalHDRPercentileRanks reduced, List<InternalHD
         return InternalHDRPercentileRanks::new;
     }
 
+    @Override
+    protected NamedXContentRegistry xContentRegistry() {
+        return new NamedXContentRegistry(singletonList(
+                new NamedXContentRegistry.Entry(
+                        Aggregation.class,
+                        new ParseField(InternalHDRPercentileRanks.NAME),
+                        (parser, context) -> ParsedHDRPercentileRanks.fromXContent(parser, (String) context))));
```

Comment:
I would insist on doing it now. Adding a line to an existing method for each agg is going to cost less than having to do it for all aggs afterwards. Any reason why we should postpone this? are we not sure it is the way to go?